src/main/java/ca/uhn/fhir/jpa/starter/service/TusService.java

```diff
@@ -0,0 +1,119 @@
+		List<String> subDirectories = getSubDirectories(appProperties.getImage_path() + File.separator + "uploads");
+		if(!subDirectories.isEmpty()){
+			for (String subDirectory: subDirectories){
+				String uploadUrl = tusServerProperties.getContextPath() + "/" + subDirectory;
+				transferImagesToFinalStorage(uploadUrl);
+			}
+		}
+	}
+
+	private void transferImagesToFinalStorage(String uploadUrl) throws TusException, IOException{
+		try{
+			InputStream inputStream = tusFileUploadService.getUploadedBytes(uploadUrl);
+			String fileName = new String(org.apache.commons.codec.binary.Base64.decodeBase64(tusFileUploadService.getUploadInfo(uploadUrl).getEncodedMetadata().split(" ")[1]), Charsets.UTF_8);
+
+			// Use ByteArrayOutputStream to collect all the bytes from the InputStream.
+			ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+
+			// Define a buffer to read data in chunks.
+			byte[] buffer = new byte[1024]; // You can adjust the buffer size as per your requirement.
+
+			int bytesRead;
+			while ((bytesRead = inputStream.read(buffer)) != -1) {
+				// Write the bytesRead number of bytes to the ByteArrayOutputStream.
+				byteArrayOutputStream.write(buffer, 0, bytesRead);
+			}
+
+			// Close the InputStream and ByteArrayOutputStream when done reading.
+			inputStream.close();
+			byteArrayOutputStream.close();
+
+			// Get the complete byte array from the ByteArrayOutputStream.
+			byte[] completeByteArray = byteArrayOutputStream.toByteArray();
+			BufferedImage image = byteArrayToBufferedImage(completeByteArray);
+			saveImageToFile(image, appProperties.getImage_path(), fileName);
+			tusFileUploadService.deleteUpload(uploadUrl);
+		} catch (FileNotFoundException e){
+			e.printStackTrace();
+		}
+	}
+
+	public static List<String> getSubDirectories(String directoryPath) {
+		List<String> subDirectories = new ArrayList<>();
+
+		File rootDirectory = new File(directoryPath);
+		if (rootDirectory.exists() && rootDirectory.isDirectory()) {
+			File[] files = rootDirectory.listFiles();
+			if (files != null) {
+				for (File file : files) {
+					if (file.isDirectory()) {
+						subDirectories.add(file.getName());
+					}
+				}
+			}
+		}
+		return subDirectories;
+	}
+
+	private static BufferedImage byteArrayToBufferedImage(byte[] byteArrayData) throws IOException {
+		ByteArrayInputStream inputStream = new ByteArrayInputStream(byteArrayData);
+		return ImageIO.read(inputStream);
+	}
+
+	private static void saveImageToFile(BufferedImage image, String folderPath, String fileName) throws IOException {
+		Path outputPath = Paths.get(folderPath, fileName);
+		File outputFile = outputPath.toFile();
+
+		// Create parent directories if they don't exist
+		Files.createDirectories(outputPath.getParent());
+		if(outputFile.exists()){
+			System.out.println("File already exists with the name" + fileName);
+		}else{
+			if (image != null)
+				ImageIO.write(image, "jpeg", outputFile);
```

Comment:
return the value that is returned by ImageIO.write and delete the file only if it is true in the transferImagesToFinalStorage function